federation/pkg/federation-controller/sync/controller.go

```diff
@@ -490,10 +490,13 @@ func syncToClusters(clustersAccessor clustersAccessorFunc, operationsAccessor op
 		if !ok {
 			glog.Fatalf("Adapter for kind %q does not properly implement SchedulingAdapter.", kind)
 		}
-		err = schedulingAdapter.UpdateFederatedStatus(obj, schedulingInfo.Status)
-		if err != nil {
-			runtime.HandleError(fmt.Errorf("adapter.UpdateFinished() failed on adapter for %s %q: %v", kind, key, err))
-			return statusError
+		if schedulingInfo != nil {
```

Comment:
I did put this check while changing the params to/from schedule adapter to interfaces which passes pointer.
You are right, this check is not at all needed as none of the current implementations actually pass a null. Removed it!